### ISSUE TITLE
Delimiter doesn't vanish when a module is stopped

### ISSUE THREAD

**Issue by paniash** [torrinfail/dwmblocks#33]
I was running an old build of dwmblocks where I had my music module (through mpd) as my first module in `blocks.h`. It looked something like this:
![play-old](https://user-images.githubusercontent.com/59497618/91555253-43a79b00-e94e-11ea-8ee5-dd59605e329b.png)

However, when I stopped (not paused) my music, the delimiter right next to it would vanish, looking like this:
![stop-old](https://user-images.githubusercontent.com/59497618/91555339-75206680-e94e-11ea-9324-64fe76a4aa3d.png)

I really like this behaviour and recently I updated my build to be current with this repo, since my old build did not support multicharacter delimiter. Everything works fine but when I stop my music module, the delimiter right next to it stays in place unlike the older build:
![stop-new](https://user-images.githubusercontent.com/59497618/91555460-b0bb3080-e94e-11ea-924e-f6a60e479448.png)

I've tried looking at `dwmblocks.c` but am unable to figure out what part of the code is responsible for this behaviour. I know this isn't much of an issue but I'd be glad if you could help me rectify it. Thanks in advance!

Here's the old `dwmblocks.c`:

```
#include<stdlib.h>
#include<stdio.h>
#include<string.h>
#include<unistd.h>
#include<signal.h>
#include<X11/Xlib.h>
#define LENGTH(X)               (sizeof(X) / sizeof (X[0]))
#define CMDLENGTH		80

typedef struct {
	char* icon;
	char* command;
	unsigned int interval;
	unsigned int signal;
} Block;
void sighandler(int num);
void getcmds(int time);
#ifndef __OpenBSD__
void getsigcmds(int signal);
void setupsignals();
void sighandler(int signum);
#endif
int getstatus(char *str, char *last);
void setroot();
void statusloop();
void termhandler(int signum);


#include "blocks.h"

static Display *dpy;
static int screen;
static Window root;
static char statusbar[LENGTH(blocks)][CMDLENGTH] = {0};
static char statusstr[2][256];
static int statusContinue = 1;
static void (*writestatus) () = setroot;

//opens process *cmd and stores output in *output
void getcmd(const Block *block, char *output)
{
	strcpy(output, block->icon);
	char *cmd = block->command;
	FILE *cmdf = popen(cmd,"r");
	if (!cmdf)
		return;
	char c;
	int i = strlen(block->icon);
	fgets(output+i, CMDLENGTH-i, cmdf);
	i = strlen(output);
	if (delim != '\0' && --i)
		output[i++] = delim;
	output[i++] = '\0';
	pclose(cmdf);
}

void getcmds(int time)
{
	const Block* current;
	for(int i = 0; i < LENGTH(blocks); i++)
	{	
		current = blocks + i;
		if ((current->interval != 0 && time % current->interval == 0) || time == -1)
			getcmd(current,statusbar[i]);
	}
}

#ifndef __OpenBSD__
void getsigcmds(int signal)
{
	const Block *current;
	for (int i = 0; i < LENGTH(blocks); i++)
	{
		current = blocks + i;
		if (current->signal == signal)
			getcmd(current,statusbar[i]);
	}
}

void setupsignals()
{
	for(int i = 0; i < LENGTH(blocks); i++)
	{	  
		if (blocks[i].signal > 0)
			signal(SIGRTMIN+blocks[i].signal, sighandler);
	}

}
#endif

int getstatus(char *str, char *last)
{
	strcpy(last, str);
	str[0] = '\0';
	for(int i = 0; i < LENGTH(blocks); i++)
		strcat(str, statusbar[i]);
	str[strlen(str)-1] = '\0';
	return strcmp(str, last);//0 if they are the same
}

void setroot()
{
	if (!getstatus(statusstr[0], statusstr[1]))//Only set root if text has changed.
		return;
	Display *d = XOpenDisplay(NULL);
	if (d) {
		dpy = d;
	}
	screen = DefaultScreen(dpy);
	root = RootWindow(dpy, screen);
	XStoreName(dpy, root, statusstr[0]);
	XCloseDisplay(dpy);
}

void pstdout()
{
	if (!getstatus(statusstr[0], statusstr[1]))//Only write out if text has changed.
		return;
	printf("%s\n",statusstr[0]);
	fflush(stdout);
}


void statusloop()
{
#ifndef __OpenBSD__
	setupsignals();
#endif
	int i = 0;
	getcmds(-1);
	while(statusContinue)
	{
		getcmds(i);
		writestatus();
		sleep(1.0);
		i++;
	}
}

#ifndef __OpenBSD__
void sighandler(int signum)
{
	getsigcmds(signum-SIGRTMIN);
	writestatus();
}
#endif

void termhandler(int signum)
{
	statusContinue = 0;
	exit(0);
}

int main(int argc, char** argv)
{
	for(int i = 0; i < argc; i++)
	{	
		if (!strcmp("-d",argv[i]))
			delim = argv[++i][0];
		else if(!strcmp("-p",argv[i]))
			writestatus = pstdout;
	}
	signal(SIGTERM, termhandler);
	signal(SIGINT, termhandler);
	statusloop();
}

```

**Comment by torrinfail** [comment]
Looking at the old code just very quickly I'm not sure how the old behavior was happening, but that is the desired behavior so I fixed it in the new version with commit 13c7700. How it works now is that it wont add the delimiter if both the icon from blocks.h and the output of the command are empty. So if that block has an icon then the delimiter will still be drawn. But in your example it appears your mpd block is entirely from output so this should fix your issue. Thanks for the clear explanation and the pictures. They were very helpful!

**Comment by paniash** [comment]
@torrinfail Thank you very much for rectifying the issue! And also thanks for lot for creating this wonderful piece of software!
